### PR TITLE
feat(python): adds a token swap example

### PR DESCRIPTION
**Pull request by elmurci**
## High Level Overview of Change

Token Swap example.

### CODE REVIEW

File: python/examples/token_swap/README.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+Token swaps are useful when two users agree on an exchange and remove the need to trade via a DEX or an AMM.
+
+In this example, the swap is a 2-step process:
+
+1. `UserA` and `UserB` agree on a swap by which `UserA` will give x `A` tokens to `UserB` and `UserB` will give y `B` tokens to `UserA`.
+2. `UserA` initiates the trade on the ledger with the pre agreed conditions via a `TokenSwapPropose` transaction.
+This creates a `TokenSwap` object in the ledger with a `TokenSwapId`.
+3. `UserB` validates the trade and accepts it (if the conditions are the pre agreed ones) by sending a `TokwnSwapAccept` transaction with the given `TokenSwapId`.
+4. If `TokwnSwapAccept` transaction is successful, the swap will be executed atomically.
```
Comment (mvadari) [inline python/examples/token_swap/README.md]:
```suggestion
3. `UserB` validates the trade and accepts it (if the conditions are the pre agreed ones) by sending a `TokenSwapAccept` transaction with the given `TokenSwapId`.
4. If `TokenSwapAccept` transaction is successful, the swap will be executed atomically.
```

File: python/examples/token_swap/README.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+### TokenSwapPropose transaction
+
+A **`TokenSwapPropose`** transaction may have the following fields:
+
+| Field Name                 | Required? | JSON Type         | Internal Type   |
+| -------------------------- | --------- | ----------------- | --------------- |
+| `AccountOther`             | ✔️         | `string`          | `ACCOUNT`       |
+| `Amount`                   | ✔️         | `Currency Amount` | `AMOUNT`        |
+| `AmountOther`              | ✔️         | `Currency Amount` | `AMOUNT`        |
```
Comment (mvadari) [inline python/examples/token_swap/README.md:50]: I added `sfAmount2` btw

File: python/examples/token_swap/token_swap.py [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+        object_format=[
+            (sf_account, soeREQUIRED),
+            (sf_amount, soeREQUIRED),
+            (sf_token_swap_id, soeREQUIRED),
+            (sf_amount_other, soeREQUIRED),
+            (sf_account_other, soeREQUIRED),
+            (sf_owner_node, soeREQUIRED),
+        ],
+        is_deletion_blocker=True,
+        # visit_entry_xrp_change=visit_entry_xrp_change_token_swap,
```
Comment (mvadari) [inline python/examples/token_swap/token_swap.py]: Is there a reason this is commented out?

Comment (elmurci) [reply python/examples/token_swap/token_swap.py]: no, my bad

File: python/examples/token_swap/README.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
+> [!WARNING]  
+> This example is not production ready and it is only for demonstration purposes.
+
```
Comment (6309304695) [inline python/examples/token_swap/README.md:3]: SCP-004v

**Review by 6309304695 — APPROVED** [review]

**Review by 6309304695 — CHANGES_REQUESTED** [review]
python/examples/token_swap/token_swap.py

**Review by 6309304695 — APPROVED** [review]
python/examples/token_swap/token_swap.py

**Pull request closed without merging**